### PR TITLE
Describe Sponge properties in the Spongineer section

### PR DESCRIPTION
**Pull request by Grauldon**
[SpongeDocs](https://github.com/SpongePowered/SpongeDocs/issues/796)

I have updated the [Becoming an Expert Spongineer](https://docs.spongepowered.org/stable/en/server/spongineer/index.html#becoming-an-expert-spongineer) section of SpongeDocs to include a page explaining the Sponge properties. The page consists of a brief description, how to use them, and a table listing them.

Signed-off-by: Grauldon <17791090+Grauldon@users.noreply.github.com>

### CODE REVIEW

File: source/server/spongineer/properties.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+| **timings.bypassMax**              | Allows for bypassing the max number of frames to send to the Aikar's timing     |   
+|                                    | API. Setting this will not permit bypassing the limit unless the API is         |  
+| (any value that exists)            | configured to permit it.                                                        |
++------------------------------------+---------------------------------------------------------------------------------+
+|                                    | Adds a StackTrace to each ``StackFrame`` for when either the Sponge             |
+| **sponge.debugcauseframes**        | implementation or a plugin potentially corrupts the ``CauseStackManager`` 's    |
+|                                    | ``StackFrame`` s, whether it's a frame being popped out of order, or popping    |
+| (boolean, default: false)          | an already popped frame. This is useful to find out where the frame was         | 
+|                                    | created.                                                                        |
```
Comment (gabizou) [inline source/server/spongineer/properties.rst:41]: Has the potential to increase performance costs due to the stacktrace creation for every frame in the duration of the game's lifecycle.

File: source/server/spongineer/properties.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
++------------------------------------+---------------------------------------------------------------------------------+
+|                                    | Similar to the ``StackFrame`` exception creation, this tracks the stack trace   |
+|                                    | for each ``BlockTransaction`` that is captured for a block being changed, a     | 
+| **sponge.debugProxyChanges**       | neighbor notification request, or a ``TileEntity`` change. Useful to find out   |
+|                                    | where each transaction is created, potentially adding more information if there |
+| (boolean, default: false)          | is an exception thrown during the processing of said transactions. Should       |
+|                                    | always be included for logs when reporting issues involving these sort of       |
+|                                    | transactions.                                                                   |
```
Comment (gabizou) [inline source/server/spongineer/properties.rst:49]: Likewise, has some performance cost in potentially intensive environments due to the exception creation for every potential `Entity`/`TileEntity`/`Block` being ticked in the game.

File: source/server/spongineer/properties.rst (hunk 1 above)
Comment (Grauldon) [reply source/server/spongineer/properties.rst:41]: I chose a warning block over additional bold-face text in the table. Let me know if adding to the table was the original intent.

File: source/server/spongineer/properties.rst (hunk 2 above)
Comment (Grauldon) [reply source/server/spongineer/properties.rst:49]: I chose a warning block over additional bold-face text in the table. Let me know if adding to the table was the original intent.

File: source/server/spongineer/properties.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Properties Available
+====================
+
++------------------------------------+---------------------------------------------------------------------------------+
+| Property                           | Description                                                                     |
++====================================+=================================================================================+
+|                                    | Forces all boolean "performance" checks for common SpongeAPI events to be       |  
+| **sponge.shouldFireAll**           | forcibly created and fired, regardless whether there is a registered event      |
+|                                    | listener for the desired event (Sponge will normally not construct and throw a  |
+| (boolean, default:false)           | ``ChangeBlockEvent`` if there's no listeners for that event or it's sub         |
```
Comment (Inscrutable) [inline source/server/spongineer/properties.rst]: Plural/singular? Either use `there are no listeners` or `there's no listener`.

File: source/server/spongineer/properties.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+|                                    | always be included for logs when reporting issues involving these sort of       |
+|                                    | transactions.                                                                   |
++------------------------------------+---------------------------------------------------------------------------------+
+| **sponge.filter.debug**            | Writes out classes that are generated by the filter generator to the            |
+|                                    | ``.sponge.debug.out`` folder, named after the class they are altering.          |
+| (boolean, default: false)          |                                                                                 |
++------------------------------------+---------------------------------------------------------------------------------+
+| **sponge.print_all_catalog_types** | Prints out all registered ``CatalogType`` s and their ``String`` id's at the    | 
+|                                    | end of startup, useful for finding out what id's are registered from vanilla,   |
```
Comment (Inscrutable) [inline source/server/spongineer/properties.rst]: Should be a semicolon (or full stop and new sentence) after `startup`. Or add `which is`

File: source/server/spongineer/properties.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+| (boolean, default: false)          | mods, and plugins for all created ``CatalogType`` s.                            |
++------------------------------------+---------------------------------------------------------------------------------+
+|                                    | Prints out a very spam filled log for every block captured that is processed,   |
```
Comment (Inscrutable) [inline source/server/spongineer/properties.rst]: Do we have better terminology than `spam-filled`? e.g. cluttered, busy, crowded, content-rich, verbose, ...

File: source/server/spongineer/properties.rst [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+|                                    | implementation, and potentially plugins, when dealing with                      |
+|                                    | ``ChangeBlockEvents`` and their processing.                                     |
++------------------------------------+---------------------------------------------------------------------------------+
+
+.. warning::
+   **sponge.debugcauseframes** has the potential to increase performance costs due to the stack trace creation for
+   every frame in the duration of the game's lifecycle.
+
+   **sponge.debugProxyChanges** has the same potential to increase performance costs in potentially intensive 
+   environments due to the exception creation for every potential ``Entity`` / ``TileEntity`` / ``Block`` being ticked
```
Comment (Inscrutable) [inline source/server/spongineer/properties.rst]: If you remove the `the` (before `exception`) it becomes clearer, imho.

**Review by Inscrutable — COMMENTED** [review]
Looking good. Just the mandatory minor quibbles.
I guess it looks more organized as a table than a definition list.

File: source/server/spongineer/properties.rst (hunk 3 above)
Comment (Grauldon) [reply source/server/spongineer/properties.rst]: Changed to `there are no listeners`.

File: source/server/spongineer/properties.rst (hunk 4 above)
Comment (Grauldon) [reply source/server/spongineer/properties.rst]: Added `which is`

File: source/server/spongineer/properties.rst (hunk 5 above)
Comment (Grauldon) [reply source/server/spongineer/properties.rst]: Changed `spam filled` to `verbose log entry`.

File: source/server/spongineer/properties.rst (hunk 6 above)
Comment (Grauldon) [reply source/server/spongineer/properties.rst]: Removed `the` .

**Comment by Grauldon** [comment]
I didn't think of a definition list. I jumped to a table based on experience and comments on the issue. I will try a definition list before I commit the other changes.

**Comment by Grauldon** [comment]
Decided to push with the table instead of a definition list.

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/df75c0c/.

Here are some links to the pages that were modified:

- added: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/df75c0c/server/spongineer/properties.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/df75c0c/server/spongineer/index.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/808#issuecomment-500544506), not to the direct url to the preview._

**Comment by Inscrutable** [comment]
Does this need the **WIP** tag any more? It looks done to me.

**Comment by Grauldon** [comment]
I removed it. I was waiting to ensure no more changes were requested.

**Pull request merged**